odbc/src/api.rs

```diff
@@ -431,8 +473,54 @@ pub extern "C" fn SQLForeignKeysW(
 }
 
 #[no_mangle]
-pub extern "C" fn SQLFreeHandle(_handle_type: HandleType, _handle: Handle) -> SqlReturn {
-    unimplemented!()
+pub extern "C" fn SQLFreeHandle(handle_type: HandleType, handle: Handle) -> SqlReturn {
+    unsafe {
+        match handle_type {
+            // By making Boxes to the types and letting them go out of
+            // scope, they will be dropped.
+            HandleType::Env => {
+                let env = Box::from_raw(handle as *mut EnvHandle);
```

Comment:
IIUC, this will result in UB if we interpret the handle as the wrong type (which could happen if the caller lies about `handle_type`). Is that allowed, or do we need to explicitly return `SQL_INVALID_HANDLE`? The state transition tables seem to suggest it is the latter, but also possibly that it's handled by the DM?

Reply:
It seems we should handle it. I know how to make this work. Thanks for pointing this out.

Reply:
Ok, this finally works! It also returns `INVALID_HANDLE` if "super handles" are not passed for `Statements` and `Connections`.